### PR TITLE
Fix for Preferred Language refresh bug in Localization process related to User Preferences Page [13653]

### PR DESCRIPTION
**Pull request by hujambo-dunia**
For detailed explanation, see this closed PR (closed due to 'bad merge'): https://github.com/galaxyproject/galaxy/pull/13825
Issue: #13653 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: client/src/utils/localization.js [hunk 1]
```diff
@@ -42,6 +24,56 @@ define(["i18n!nls/locale"], function (localeStrings) {
...
+
+        sessionStorage.setItem("currentLocale", locale);
+    };
+    // =============================================================================
+    /** Simple string replacement localization. Language data from src/nls */
+
+    localize._getUserLocale = function _getUserLocale(user, config) {
+        let locale = null;
+        if (Object.prototype.hasOwnProperty.call(localeStrings, "__root")) {
+            //console.debug( 'amdi18n+webpack localization for ' + locale + ' loaded' );
```
Comment (bgruening) [inline client/src/utils/localization.js]: @hujambo-dunia do you plan to clean this up before merging?

Comment (hujambo-dunia) [reply client/src/utils/localization.js]: @bgruening yes - for one thing, there were a bunch of comments left by prior developer which I would like to remove.

Comment (dannon) [reply client/src/utils/localization.js]: +1, just delete any old commented code.  We want to ditch this old plugin and swap to a more modern https://kazupon.github.io/vue-i18n/ anyway now that we're closing in on converting all primary interfaces over.

File: client/src/utils/localization.js [hunk 2]
```diff
@@ -1,47 +1,53 @@
-// eslint-disable-next-line no-undef
```
Comment (dannon) [inline client/src/utils/localization.js:1]: This one wasn't just a comment to be removed; it was an eslint directive.  If you put this one back it'll lint cleanly again.

**Pull request merged**